docs/en/ingest-management/processors/processors.asciidoc

```diff
@@ -0,0 +1,127 @@
+* sanitize data
+
+Each processor receives an event, applies a defined action to the event, and
+returns the event. If you define a list of processors, they are executed in the
+order they are defined.
+
+[source,yaml]
+----
+event -> processor 1 -> event1 -> processor 2 -> event2 ...
+----
+
+[discrete]
+[[where-valid]]
+== Where are processors valid?
+
+The processors described in this section are valid:
+
+* **Under integration settings in the Integrations UI in {kib}**. For example,
+when configuring an Nginx integration, you can define processors for a specific
+dataset under **Advanced options**. The processor in this example adds geo
+metadata to the Nginx access logs collected by {agent}:
++
+[role="screenshot"]
+image::images/add-processor.png[Screen showing how to add a processor to an integration policy]
++
+NOTE: Some integrations do not currently support processors.
+
+* **Under input configuration settings for standalone {agent}s**. For example:
++
+[source,yaml]
+----
+inputs:
+  - type: logfile
+    use_output: default
+    data_stream:
+      namespace: default
+    streams:
+      - data_stream:
+          dataset: nginx.access
+          type: logs
+        ignore_older: 72h
+        paths:
+          - /var/log/nginx/access.log*
+        tags:
+          - nginx-access
+        exclude_files:
+          - .gz$
+        processors:
+          - add_host_metadata:
+              cache.ttl: 5m
+              geo:
+                name: nyc-dc1-rack1
+                location: '40.7128, -74.0060'
+                continent_name: North America
+                country_iso_code: US
+                region_name: New York
+                region_iso_code: NY
+                city_name: New York
+          - add_locale: null
+----
+
+//REVIEWERS: Does this config look correct? I modified the policy created
+//by Fleet, but not sure if I stripped out anything important.
+
+You can define processors that apply to all inputs defined in the configuration,
+or specific inputs.
+
+[discrete]
+[[limitations]]
+== What are some limitations of using processors?
+
+Processors have the following limitations.
+
+* Cannot enrich events with data from {es} or other custom data
+sources.
+* Cannot process data after it's been converted to the Elastic Common
+Schema (ECS) because the conversion is performed by {es} ingest pipelines.
+* May break integration ingest pipelines in {es} if the user-defined processing
+removes or alters fields expected by ingest pipelines.
+
+//REVIEWERS: Need more input into the last item in this list to make it more
+//accurate.
+
+[discrete]
+[[processing-options]]
+== What other options are available for processing data?
+
+//REVIEWERS: I'd like to consider reusing a portion of the diagram we created
+//for the ingest cloud docs. I think the decision tree is more useful than
+//a bunch of words, and it sends users off to learn more. WDYT?
+
+The {stack} provides several options for processing data collected by {agent}.
```

Comment:
@colleenmcginnis What do you think of providing a pared down version of your diagram here? I think it's pretty useful for folks who have already committed to using Elastic Agent, but discover they need to do additional processing. If you think it's a good idea, let me know the best way to proceed.

Reply:
Nothing here that I would want to replace "Do you need to process your data?" with something like "What do you need to do?"

Reply:
I don't have a strong opinion here. I think we could create a variation of the diagram, but I also think a question-answer format would also work and might be easier to maintain. Would you need the links to Beats-related docs? A simple question-answer format could look something like:

>**Do you need to sanitize or enrich raw data at the source?**
>Use an [Agent processor]().
>
>**Do you need to convert data to ECS, normalize field data, or enrich incoming data?**
>Use [ingest pipelines]().
>
>**Do you need to define or alter the schema at query time?**
>Use [runtime fields]().
>
>**Do you need to do something else with your data?**
>Use [Logstash plugins]().

Reply:
That's a good point! A branching diagram is probably overkill here since there are fewer decision points for folks using Elastic Agent. It's probably _not_ worth the extra maintenance.